onnxruntime/test/perftest/ort_test_session.cc

```diff
@@ -41,13 +41,48 @@ OnnxRuntimeTestSession::OnnxRuntimeTestSession(Ort::Env& env, std::random_device
 #endif
   } else if (provider_name == onnxruntime::kCudaExecutionProvider) {
 #ifdef USE_CUDA
-    OrtCUDAProviderOptions cuda_options{
-        0,
-        static_cast<OrtCudnnConvAlgoSearch>(performance_test_config.run_config.cudnn_conv_algo),
-        std::numeric_limits<size_t>::max(),
-        0,
-        !performance_test_config.run_config.do_cuda_copy_in_separate_stream};
-    session_options.AppendExecutionProvider_CUDA(cuda_options);
+    const auto& api = Ort::GetApi();
+
+    OrtCUDAProviderOptions* cuda_options;
+    Ort::ThrowOnError(api.CreateCUDAProviderOptions(&cuda_options));
+    std::unique_ptr<OrtCUDAProviderOptions, decltype(api.ReleaseCUDAProviderOptions)> rel_cuda_options(cuda_options, api.ReleaseCUDAProviderOptions);
+
+    std::vector<const char*> cuda_options_keys{"device_id", "arena_extend_strategy", "cuda_mem_limit",
+                                               "cudnn_conv_algo_search", "do_copy_in_default_stream"};
+    std::vector<const char*> cuda_options_values;
+    cuda_options_values.reserve(5);
+
+    // device id
+    cuda_options_values.push_back("0");
+
+    // arena extend strategy
+    cuda_options_values.push_back("kNextPowerOfTwo");
+
+    // cuda mem limit
+    cuda_options_values.push_back(std::to_string(std::numeric_limits<size_t>::max()).c_str());
```

Comment:
>std::to_string(std::numeric_limits<size_t>::max()).c_str() [](start = 34, length = 58)

this appears to store a pointer to memory within a temporary variable.

Reply:
Gosh - yes, thanks for spotting that. I created a new var that holds the string equivalent of std::numeric_limits<size_t>::max() and its lifetime should now be the lifetime of this method and passed in the pointer to memory held by it to cuda_options_values.